modules/common/log/default.nix

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{lib, ...}:
+with lib; {
+  options.ghaf.logging.client.enable = lib.mkOption {
+    description = ''
+      Enable logging client service. Currently we have grafana alloy
+      running as client which will upload system journal logs to
+      grafana loki running in admin-vm
+    '';
+    type = lib.types.bool;
+    default = false;
+  };
+
+  options.ghaf.logging.server.enable = lib.mkOption {
+    description = ''
+      Enable logging server option will add functionality in
+      device to upload logs to remote server.
+    '';
+    type = lib.types.bool;
+    default = false;
+  };
+
+  options.ghaf.logging.port = lib.mkOption {
+    description = ''
+      Assign port value to grafana alloy process service running in
+      admin-vm. Using this port value firewall allowed TCP port
+      policy will be created Ghaf internally to grafana loki
+      (by default in admin-vm). No external network firewall policies
+      are created
+    '';
+    type = lib.types.int;
+    default = 3100;
+  };
+
+  options.ghaf.logging.listener.port = lib.mkOption {
+    description = ''
+      Assign port value to grafana alloy source service running in
+      different clients. Using this port value firewall allowed TCP port
+      policy will be created Ghaf internally to grafana loki
+      (by default in admin-vm). No external network firewall policies
+      are created
+    '';
+    type = lib.types.int;
+    default = 9999;
+  };
+
+  options.ghaf.logging.server.address = lib.mkOption {
+    description = ''
+      This is the IP of grafana cloud server. If server logging is enabled
+      we need to specify the IP where device logs will be uploaded.
+    '';
+    type = lib.types.str;
+    default = "x.x.x.x";
```

Comment:
I would remove invalid default and leave the specific declaration to targets.

Reply:
Removed default value.